chore: Improve documentations (#12)
saucelabs/node-testcomposer: README.md

# Usage

> ## Initial TestComposer Client

Reviewer: ```suggestion
## Initialize TestComposer Client
```

```javascript
const { TestComposer } = require('@saucelabs/testcomposer');

const client = new TestComposer({
  region: 'us-west-1',
  username: process.env.SAUCE_USERNAME,
  accessKey: process.env.SAUCE_ACCESS_KEY,
  headers: {'User-Agent': `your-fancy-reporter/1.2.3`}
});
```

> ## Create Reports for JavaScript Batch Frameworks

Reviewer: Is referring to "Batch Frameworks" here helpful? "Batch" is a name we use internally and probably doesn't connote anything meaningful to most users.

Author: What about just `JavaScript Frameworks`?

  endTime: new Date().toISOString(),
  browserName: "Chrome",
  browserVersion: "105",
  framework: "playwright", // alternatives: cypress, playwright, testcafe
  frameworkVersion: "1.25.0",
  platformName: "Windows 11"
});
console.log(job.id); // the job ID
console.log(job.url); // the full URL of the job
```

### Upload log for JavaScript Batch Frameworks

> The minimum `sauce-test-report.json` file to fulfill the web UI:

Reviewer: "Fulfill the web UI" is a bit terse and awkward. Can you describe a bit more clearly what the file is for?

And the phrasing `The minimum 'sauce-test-report.json' file` makes it sound like the contents of the snippet verbatim below is the "minimum" report required. But I'm guessing what you mean is at a minimum, a file called `sauce-test-report.json` is required?

Should we then also refer to the [sauce-json-reporter](https://github.com/saucelabs/sauce-json-reporter-js) lib for generating the reports?

  endTime: new Date().toISOString(),
  browserName: "Chrome",
  browserVersion: "105",
  framework: "playwright", // alternatives: cypress, playwright, testcafe
  frameworkVersion: "1.25.0",
  platformName: "Windows 11"
});
console.log(job.id); // the job ID
console.log(job.url); // the full URL of the job
```

> ## Upload log for JavaScript Batch Frameworks

Reviewer 2: Imho this should be a subtopic under https://github.com/saucelabs/node-testcomposer/pull/12/files#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5R19

Author: fixed 👍

> The minimum `sauce-test-report.json` file to fullfill the web UI:

Reviewer 2: Typo, but also, what does "fulfill the web UI" mean?
```suggestion
The minimum `sauce-test-report.json` file to fulfill the web UI:
```

```json
{
  "status": "passed",
  "attachments": [],
  "suites": [
    {
      "name": "somegroup",
      "status": "passed",
      "metadata": {},
      "suites": [
        {
          "name": "somefile.test.js",
          "status": "passed",
          "metadata": {},
          "suites": [],
          "attachments": [],
          "tests": [
            {
              "name": "yay",
              "status": "passed",
              "duration": 123,
              "startTime": "2023-07-06T22:03:19.591Z",
              "attachments": [],
              "metadata": {}
            }
          ]
        }
      ],
      "attachments": [],
      "tests": []
    }
  ],
  "metadata": {}
}
```
To upload `sauce-test-report.json`:

```javascript
const uploads = await client.uploadAssets(
  job.id,
  [{
    filename: "log.json",

>     data: fs.createReadStream("log.json")

Reviewer 2: Aren't you [trying to upload](https://github.com/saucelabs/node-testcomposer/pull/12/files#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5R77) "sauce-test-report.json"?

Author: ah.. Nice catch!

  }]
);
// Upon success, the array `uploads.uploaded` will equal in length to the number of assets you intended to upload.
// Individual assets that fail to upload will be reported via `uploads.errors`.
```

## Create Reports for Selenium

```javascript
const job = await client.createReport({
  name: "My Fancy Job!",
  passed: true,
  startTime: new Date().toISOString(),
  endTime: new Date().toISOString(),
  browserName: "Chrome",
  browserVersion: "105",
  framework: "webdriver",
  platformName: "Windows 11"
});
console.log(job.id); // the job ID
console.log(job.url); // the full URL of the job
```

> ## Upload Log for Selenium job

Reviewer 2: Subtopic 🙏

> The minimum `log.json` file to fullfill the web UI:

Reviewer 2: Same as above.
```suggestion
The minimum `log.json` file to fulfill the web UI:
```

```json
[
	{
    "screenshot": null,
    "suggestion_values": [

    ],
    "start_time": 1688673799.516,
    "request": {
      "using": "css selector",
      "value": "[id=\"onetrust-accept-btn-handler\"]"
    },
    "result": {
      "message": "no such element",

>       "error": "no such element"

Reviewer 2: You say this is the minimal input, but can't result be empty? Wouldn't that make this optional?

Would be also good to add some description as to what these fields mean.
Like `"statusCode": 1`. What values are acceptable?

  endTime: new Date().toISOString(),
  browserName: "Chrome",
  browserVersion: "105",
  framework: "playwright", // alternatives: cypress, playwright, testcafe
  frameworkVersion: "1.25.0",
  platformName: "Windows 11"
});
console.log(job.id); // the job ID
console.log(job.url); // the full URL of the job
```

### Upload Log

> The minimum `sauce-test-report.json` file, which can be generated by [sauce-json-reporter](https://github.com/saucelabs/sauce-json-reporter-js/), to populate the test case list on Sauce Labs web UI:

Reviewer: ```suggestion
In order to view your test results in the SauceLabs web app, you must upload a test report file along with the report created in the step above. You can use the [sauce-json-reporter](https://github.com/saucelabs/sauce-json-reporter-js) library to generate the test report programatically or manually create the json file according to the report schema defined [here](https://github.com/saucelabs/sauce-json-reporter-js/blob/main/api/schema.json).
```
```

> In order to view your test results in the SauceLabs web app, you must upload a test report file along with the report created in the step above. You can use the [sauce-json-reporter](https://github.com/saucelabs/sauce-json-reporter-js) library to generate the test report programatically or manually create the json file according to the report schema defined [here](https://github.com/saucelabs/sauce-json-reporter-js/blob/main/api/schema.json).

Reviewer 2: ```suggestion
In order to view your test results in the SauceLabs web app, you must upload a test report file along with the report created in the step above. You can use the [sauce-json-reporter](https://github.com/saucelabs/sauce-json-reporter-js) library to generate the test report programmatically or manually create the json file according to the report schema defined [here](https://github.com/saucelabs/sauce-json-reporter-js/blob/main/api/schema.json).
```

Reviewer: Huh, I've been misspelling that my whole career, apparently. 🤦
